experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,654 @@
+import Oscar.gens, AbstractAlgebra.direct_sum, Oscar.invariant_ring
+#export ReductiveGroup, reductive_group, representation_matrix, group, mu_star, reynolds_operator, group_ideal, canonical_representation, natural_representation, vector_space_dimension, 
+#export InvariantRing, invariant_ring, gens, hilbert_ideal, derksen_ideal
+##########################
+#Reductive Groups
+##########################
+mutable struct ReductiveGroup
+    field::Field #characteristic zero. implement check? 
+    group::Tuple{Symbol, Int}
+    group_ideal::MPolyIdeal
+    reynolds_operator::Function
+    canonical_representation::AbstractAlgebra.Generic.MatSpaceElem
+    
+    function ReductiveGroup(sym::Symbol, m::Int, fld::Field) #have not decided the representation yet
+        G = new()
+        if sym != :SL
+            error("Only implemented for SLm")
+        end
+        G.field = fld
+        G.group = (sym,m)
+        G.reynolds_operator = reynolds_slm
+        R,_ = PolynomialRing(fld,"z"=> 1:m^2)
+        M = matrix(R,m,m,[0 for i in 1:m^2])
+        for i in 1:m
+            M[1:m,i] = gens(R)[m*(i-1)+1:i*m]
+        end
+        G.canonical_representation = M
+        G.group_ideal = ideal([det(M) - 1])
+        #base ring of M has to be the same as the representation matrix when that is created later.
+        return G
+    end
+    
+    function ReductiveGroup(sym::Symbol, m::Int, ring::MPolyRing) #have not decided the representation yet
+        G = new()
+        if sym != :SL
+            error("Only implemented for SLm")
+        end
+        G.field = base_ring(ring)
+        m = Int(sqrt(ngens(ring)))
```

Comment:
I think this should rather be something like this:
```suggestion
        @assert m^2 = ngens(ring))
```
seeing as `m` is an argument, too